.github/workflows/build-test-deploy.yml

```
@@ -0,0 +1,106 @@
+name: Build, test, deploy
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - "*"
+  pull_request:
+    branches:
+      - main
+
+defaults:
+  run:
+    shell: bash
+
+concurrency:
+  group: python-${{ github.ref }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read
+
+jobs:
+
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3
+      - name: Display Python information
+        run: python -c "import sys; print(sys.version)"
+      - name: Build sdist and wheel
+        run: pipx run build
+      - name: Check release tools
+        run: pipx run twine check dist/*
+      - name: Save build output
+        uses: actions/upload-artifact@v3
+        with:
+          name: dist
+          path: dist/
+
+  test:
+    needs: build
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.8", "3.9", "3.10", "3.11"]
+        install: [repo]
+        include:
+          - python-version: "3.11"
+            install: sdist
+          - python-version: "3.11"
+            install: wheel
+          - python-version: "3.11"
+            install: editable
```

Not doing anything that could possibly vary across versions.

```suggestion
        python-version: ["3"]
        install: ["repo", "editable", "sdist", "wheel"]
```